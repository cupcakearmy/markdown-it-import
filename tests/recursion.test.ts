import { describe, expect, test } from 'bun:test'
import { render } from './utils.ts'

describe('base', () => {
  test('should not be able to import itself', async () => {
    expect(() => render('self-reference.md')).toThrow(/cycles are not allowed, already parsed.*self-reference\.md/)
  })

  test('should not be able to import cycles', async () => {
    expect(() => render('cycle-a.md')).toThrow(/cycles are not allowed, already parsed.*cycle-b\.md/)
    expect(() => render('cycle-b.md')).toThrow(/cycles are not allowed, already parsed.*cycle-a\.md/)
  })

  test('import different files', async () => {
    expect(await render('rec-a.md')).toMatchSnapshot()
  })
})
