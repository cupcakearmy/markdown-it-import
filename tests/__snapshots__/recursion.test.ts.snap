// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`base import different files 1`] = `
"<h1>I am gonna import different stuff</h1>
<h2>Here is a typescript snippet</h2>
<blockquote>
<p>TypeScript is a strongly typed programming language that builds on JavaScript, giving you better tooling at any scale.</p>
</blockquote>
<pre><code class="language-ts">export function sum(a: number, b: number): number {
  return a + b
}

</code></pre>
"
`;
